packages/web-client/src/graph/permissions/permissions.ts

```diff
@@ -0,0 +1,232 @@
+import {
+  CollectionOfPermissionsWithAllowedValues,
+  DrivesPermissionsApiFactory,
+  DrivesRootApiFactory,
+  Permission,
+  RoleManagementApiFactory,
+  UnifiedRoleDefinition
+} from './../generated'
+import type { GraphFactoryOptions, GraphRequestOptions } from './../types'
+import type { GraphPermissions } from './types'
+
+export const PermissionsFactory = ({
+  axiosClient,
+  config
+}: GraphFactoryOptions): GraphPermissions => {
+  const drivesRootApiFactory = DrivesRootApiFactory(config, config.basePath, axiosClient)
+  const roleManagementApiFactory = RoleManagementApiFactory(config, config.basePath, axiosClient)
+  const drivesPermissionsApiFactory = DrivesPermissionsApiFactory(
+    config,
+    config.basePath,
+    axiosClient
+  )
+
+  return {
+    async getPermission<T extends CollaboratorShare | LinkShare>(
+      driveId: string,
+      itemId: string,
+      permId: string,
+      graphRoles: UnifiedRoleDefinition[],
+      requestOptions: GraphRequestOptions
+    ): Promise<T> {
+      const { data: permission } = await drivesPermissionsApiFactory.getPermission(
+        driveId,
+        itemId,
+        permId,
+        requestOptions
+      )
+
+      if (permission.link) {
+        return buildLinkShare({ graphPermission: permission, resourceId: itemId }) as T
+      }
+
+      return buildCollaboratorShare({
+        graphPermission: permission,
+        resourceId: itemId,
+        graphRoles: graphRoles || []
+      }) as T
+    },
+
+    async listPermissions(driveId, itemId, isSpaceRoot, graphRoles, requestOptions) {
+      let responseData: CollectionOfPermissionsWithAllowedValues
+
+      if (isSpaceRoot) {
+        const { data } = await drivesRootApiFactory.listPermissionsSpaceRoot(
+          driveId,
+          requestOptions
+        )
+        responseData = data
+      } else {
+        const { data } = await drivesPermissionsApiFactory.listPermissions(
+          driveId,
+          itemId,
+          requestOptions
+        )
+        responseData = data
+      }
+
+      const permissions = responseData.value || []
+      const allowedActions = responseData['@libre.graph.permissions.actions.allowedValues']
+      const allowedRoles = responseData['@libre.graph.permissions.roles.allowedValues']
+
+      const shares = permissions.map((permission) => {
+        if (permission.link) {
+          return buildLinkShare({ graphPermission: permission, resourceId: itemId })
+        }
+
+        return buildCollaboratorShare({
+          graphPermission: permission,
+          resourceId: itemId,
+          graphRoles: graphRoles || []
+        })
+      })
+
+      return { shares, allowedActions, allowedRoles }
+    },
+
+    async updatePermission<T extends CollaboratorShare | LinkShare>(
+      driveId: string,
+      itemId: string,
+      permId: string,
+      data: Permission,
+      isSpaceRoot: boolean,
```

Comment:
do we need the `isSpaceRoot` prop? Can't we derive that from `driveId === itemId`?

Reply:
Seems to work 👍